akka-contrib/docs/circuitbreaker.rst

```diff
@@ -0,0 +1,184 @@
+Circuit-Breaker Actor
+======================
+
+This is an alternative implementation of the [AKKA Circuit Breaker Pattern](http://doc.akka.io/docs/akka/snapshot/common/circuitbreaker.html).
+The main difference is that is intended to be used only for request-reply interactions with actor using the Circuit-Breaker as a proxy of the target one
+in order to provide the same failfast functionalities and a protocol similar to the AKKA Pattern implementation
+
+
+### Usage
+
+Let's assume we have an actor wrapping a back-end service and able to respond to `Request` calls with a `Response` object
+containing an `Either[String, String]` to map successful and failed responses. The service is also potentially slowing down
+because of the workload.
+
+A simple implementation can be given by this class::
+
+
+   object SimpleService {
+     case class Request(content: String)
+     case class Response(content: Either[String, String])
+     case object ResetCount
+   }
+
+   /**
+    * This is a simple actor simulating a service
+    * - Becoming slower with the increase of frequency of input requests
+    * - Failing around 30% of the requests
+    */
+   class SimpleService extends Actor with ActorLogging {
+     import SimpleService._
+
+     var messageCount = 0
+
+     import context.dispatcher
+
+     context.system.scheduler.schedule(1.second, 1.second, self, ResetCount)
+
+     override def receive = {
+       case ResetCount =>
+         messageCount = 0
+
+       case Request(content) =>
+         messageCount += 1
+         // simulate workload
+         Thread.sleep( 100 * messageCount )
+         // Fails around 30% of the times
+         if(Random.nextInt(100) < 70 ) {
+           sender ! Response(Right(s"Successfully processed $content"))
+         } else {
+           sender ! Response(Left(s"Failure processing $content"))
+         }
+
+     }
+   }
```

Comment:
This is a lot of code (below snippets too) to be put literally in the docs. 
Please create a file in tests which contains these, and then use `.. includecode::`
(e.g. `.. includecode:: code/docs/stream/io/StreamFileDocSpec.scala#file-source`)